### PR TITLE
timing: benchmark builds from --watch

### PR DESCRIPTION
**Pull request by maiamcc**
note that process.terminate() will leave an orphan daemon running until we put better signal handling somewhere

### CODE REVIEW

File: scripts/timing.py [hunk 1]
```diff
@@ -20,47 +22,119 @@
...
           'blorgly-backend project in your $GOPATH (`github.com/windmilleng/blorgly-backend`)')
     print()
 
     os.chdir(BLORGLY_BACKEND_DIR)
 
     cases = [
-        make_case_tilt_up_once(),
-        make_case_tilt_up_again_no_change(),
-        make_case_tilt_up_again_new_file(),
+        # make_case_tilt_up_once(),
```
Comment (jazzdan) [inline scripts/timing.py]: 👀

File: scripts/timing.py [hunk 2]
```diff
@@ -69,7 +143,7 @@ def set_tilt_up_called():
                 functools.partial(time_call, tilt_up_cmd))
 
 
-def make_case_tilt_up_again_no_change():
+def make_case_tilt_up_again_no_change() -> Case:
```
Comment (jazzdan) [inline scripts/timing.py]: types! feels good

File: scripts/timing.py [hunk 3]
```diff
@@ -40,27 +53,88 @@ def main():
...
-            c.setup()
-            timetake = c.test()
+            print('~~ RUNNING CASE: {}'.format(c.name))
+            args = []
+            kwargs = {}
+
+            print('~~~~ setup: {}'.format(c.name))
+            ret = c.setup()
+            if ret is not None:
+                args = ret[0]
```
Comment (nicks) [inline scripts/timing.py]: this API seems...weird.

specifically, the setup() and test() methods are tightly coupled, and using them correctly requires a lot of inside knowledge about how they're implemented.

possible alternatives:
1) is there a reason to have separate setup and test methods at all? i'm not sure that the separation has any upsides
2) make case a real class that keeps internal state

wdyt?

Comment (nicks) [reply scripts/timing.py]: ya, not blocking or anything

**Pull request merged**